src/main/java/net/ccbluex/liquidbounce/features/module/modules/render/SnakeGame.kt

```diff
@@ -0,0 +1,290 @@
+                snake[i].x = snake[i - 1].x
+                snake[i].y = snake[i - 1].y
+            }
+            when (lastKey) {
+                205 -> snake[0].x++
+                203 -> snake[0].x--
+                200 -> snake[0].y--
+                208 -> snake[0].y++
+            }
+            if (Mode == "Hard") {
+                for (obs in obstacles) {
+                    if (snake[0].x == obs.x && snake[0].y == obs.y) {
+                        checkHighScore()
+                        setupGame()
+                        return@handler
+                    }
+                }
+            }
+            for (i in 1 until snake.size) {
+                if (snake[i].x == snake[0].x && snake[i].y == snake[0].y) {
+                    checkHighScore()
+                    setupGame()
+                    return@handler
+                }
+            }
+        }
+    }
+
+    val onRender2D = handler<Render2DEvent> {
+        val sr = ScaledResolution(mc)
+        val w = sr.scaledWidth
+        val h = sr.scaledHeight
+        val sx = (w / 2 - FIELD_WIDTH / 2).toDouble()
+        val sy = (h / 2 - FIELD_HEIGHT / 2).toDouble()
+        for (i in 0 until 18) {
+            drawGradientRect(
+                sx.toInt() - i + 4,
+                sy.toInt() - i - 3,
+                sx.toInt() + FIELD_WIDTH + i - 4,
+                sy.toInt() + FIELD_HEIGHT + i - 4,
+                Color(0, 0, 0, 120).rgb,
+                Color(0, 0, 0, 120).rgb,
+                0f
+            )
+            drawBorder(
+                sx - i + 15,
+                sy - i + 15,
+                sx + FIELD_WIDTH + i - 15,
+                sy + FIELD_HEIGHT + i - 15,
+                Color(6, 70, 255, 120).rgb
+            )
+        }
+        drawRect(sx, sy, sx + FIELD_WIDTH, sy + FIELD_HEIGHT, Color(30, 0, 0, 0).rgb)
+        val fx = food.x * BLOCK_SIZE + sx
+        val fy = food.y * BLOCK_SIZE + sy
+        val cFood = ColorUtils.fade(Color(255, 15, 15), 1, 3)
+        drawRect(fx, fy, fx + BLOCK_SIZE, fy + BLOCK_SIZE, cFood.rgb)
+        if (Mode == "Hard" || Mode == "Normal" || Mode == "Easy") {
+            for (obs in obstacles) {
+                val ox = obs.x * BLOCK_SIZE + sx
+                val oy = obs.y * BLOCK_SIZE + sy
+                val cObs = ColorUtils.fade(Color(255, 255, 0), 1, 3)
+                drawRect(ox, oy, ox + BLOCK_SIZE, oy + BLOCK_SIZE, cObs.rgb)
+            }
+        }
+        for (i in snake.indices) {
+            val xx = snake[i].x * BLOCK_SIZE + sx
+            val yy = snake[i].y * BLOCK_SIZE + sy
+            val cc = ColorUtils.fade(Color(255, 253, 255), i, snake.size)
+            drawRect(xx, yy, xx + BLOCK_SIZE, yy + BLOCK_SIZE, cc.rgb)
+        }
+        if (snake[0].x * BLOCK_SIZE + sx >= sx + FIELD_WIDTH
+            || snake[0].x * BLOCK_SIZE + sx < sx
+            || snake[0].y * BLOCK_SIZE + sy < sy
+            || snake[0].y * BLOCK_SIZE + sy >= sy + FIELD_HEIGHT

```

Comment:
```suggestion
        val adjustedX = snake[0].x * BLOCK_SIZE
        if (adjustedX + sx >= sx + FIELD_WIDTH
            || adjustedX + sx < sx
            || adjustedX + sy < sy
            || adjustedX + sy >= sy + FIELD_HEIGHT
```
consider using a variable for this repeated expression